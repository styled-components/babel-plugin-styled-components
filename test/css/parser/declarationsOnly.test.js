import parse from '../../../src/css/parser';

describe('parse declarations only', () => {
  it('parses normal declarations', () => {
    expect(parse`
      color: red;
      text-align: center;
    `).toMatchSnapshot();
  });

  it('throws if semicolon is missing', () => {
    expect(() => parse`
      color: red;
      text-align: center
      font-family: "Operator Mono";
    `).toThrow();
  });

  it('throws if colon is missing', () => {
    expect(() => parse`
      color red;
      text-align: center;
    `).toThrow();
  });

  it('throws if property contains a whitespace', () => {
    expect(() => parse`
      color is: red;
      text-align: center;
    `).toThrow();
  });
});
