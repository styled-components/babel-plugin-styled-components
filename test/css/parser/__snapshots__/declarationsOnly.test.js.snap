exports[`parse declarations only parses normal declarations 1`] = `
Object {
  "declarations": Array [
    Object {
      "loc": Array [
        0,
        17,
      ],
      "property": Object {
        "loc": Array [
          0,
          12,
        ],
        "type": "PropertyLiteral",
        "value": "
      color",
      },
      "type": "Declaration",
      "value": Object {
        "interpolations": Array [],
        "loc": Array [
          13,
          17,
        ],
        "type": "ValueLiteral",
        "value": " red",
      },
    },
    Object {
      "loc": Array [
        18,
        43,
      ],
      "property": Object {
        "loc": Array [
          18,
          35,
        ],
        "type": "PropertyLiteral",
        "value": "
      text-align",
      },
      "type": "Declaration",
      "value": Object {
        "interpolations": Array [],
        "loc": Array [
          36,
          43,
        ],
        "type": "ValueLiteral",
        "value": " center",
      },
    },
  ],
  "loc": Array [
    0,
    48,
  ],
  "rules": Array [],
  "type": "Block",
}
`;
